BRANCHING_RELEASE_STRATEGY.md

```diff
@@ -0,0 +1,178 @@
+don't hesitate to [contact us](https://www.powsybl.org/pages/community/) to discuss it. We ensure the release of our most
+used components but this check list can evolve as users' demand does.
+
+You can also participate in our [TSC meetings](https://lists.lfenergy.org/g/powsybl-tsc/) for your voice to be heard.
+
+After a release train is published, a new date for the next release train is fixed.
+
+## Releasing a repository: a guide
+
+### Prerequisites
+
+In order to release a PowSyBl repository, you must first:
+- be a maintainer of the repository you wish to release
+- have a Sonatype JIRA account, that can be created [here](https://issues.sonatype.org/secure/Signup!default.jspa)
+- have rights to upload artefacts with the group ID `com.powsybl`; this must be achieved by having a current maintainer to create an issue on [Sonatype JIRA Dashboard](https://issues.sonatype.org/secure/Dashboard.jspa) asking to grant you these rights
+- have a PGP/GPG key to sign your release; the complete documentation is available [here](https://central.sonatype.org/pages/working-with-pgp-signatures.html)
+- configure the server in your maven settings (by default in you `~/.m2/settings.xml` file):
+```xml
+<servers>
+         ...
+         <server>
+             <id>ossrh</id>
+             <username>SONARTYPE_LOGIN</username>
+             <password>SONARTYPE_PASSWORD</password>
+         </server>
+</servers>
+- ```
+
+### Generating a release tag
+
+For the sake of the demonstration, the repository to be released will be called `powsybl-repo` below.
+
+Start by being up-to-date to your `main` branch:
+```shell
+$ cd powsybl-repo
+$ git checkout main
+$ git pull
+```
+
+Create your temporary branch preparing to the release X.Y.0 and add commits bumping to your release version then your next snapshot version.
+You can then push this branch.
+```shell
+$ git checkout -b tmp_prepare_release
+$ mvn versions:set -DnewVersion=X.Y.0
+$ git commit -s -m "Bump to vX.Y.0"
+$ mvn versions:set -DnewVersion=X.Y+1.0-SNAPSHOT
+$ git commit -s -m "Bump to vX.Y+1.0-SNAPSHOT"
+$ git push -u origin tmp_prepare_release
+```
+
+Create a pull request from your temporary branch into the `main` branch and tag another maintainer as a reviewer so they can approve it.
+Once it is approved, then merge it with the option `Rebase and merge`.
+
+You can then create a Release note and tag via the GitHub UI, using the hash of the corresponding commit (bumping to vX.Y.0). The tag must respect the pattern `vX.Y.0`
+Please make sure that your release note is comprehensive to all new features and bug fixes of the release and that the migration guide has been updated if necessary.
+
+### Publishing a release
+
+**Before creating your local release, please make sure that your local repository state is clean.**
+
+On your repository, checkout to the release tag. You can then package and deploy your release:
+```shell
+$ git checkout tags/vX.Y.0
```

Comment:
We could insert a repo purge? and ask for doing this in a temporary folder where we'd clone the repo to have a clean state.
```suggestion
$ git checkout tags/vX.Y.0
$ mvn dependency:purge-local-repository
```